applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.cpp

```diff
@@ -213,28 +212,50 @@ Vector& ElasticIsotropic3D::CalculateValue(
     if (rThisVariable == STRAIN ||
         rThisVariable == GREEN_LAGRANGE_STRAIN_VECTOR ||
         rThisVariable == ALMANSI_STRAIN_VECTOR) {
-        this->CalculateCauchyGreenStrain( rParameterValues, rValue);
+
+        Flags& r_flags = rParameterValues.GetOptions();
+
+        // Previous flags saved
+        const bool flag_const_tensor = r_flags.Is( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR );
+        const bool flag_stress = r_flags.Is( ConstitutiveLaw::COMPUTE_STRESS );
+
+        // Set flags to only compute the strain
+        r_flags.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, false);
+        r_flags.Set(ConstitutiveLaw::COMPUTE_STRESS, false);
+
+        ElasticIsotropic3D::CalculateMaterialResponsePK2(rParameterValues);
+        rValue = rParameterValues.GetStrainVector();
+
+        // Previous flags restored
+        r_flags.Set( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, flag_const_tensor );
+        r_flags.Set( ConstitutiveLaw::COMPUTE_STRESS, flag_stress );
+
     } else if (rThisVariable == STRESSES ||
         rThisVariable == CAUCHY_STRESS_VECTOR ||
         rThisVariable == KIRCHHOFF_STRESS_VECTOR ||
         rThisVariable == PK2_STRESS_VECTOR) {
-        // Get Values to compute the constitutive law:
+
         Flags& r_flags = rParameterValues.GetOptions();
 
         // Previous flags saved
         const bool flag_const_tensor = r_flags.Is( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR );
         const bool flag_stress = r_flags.Is( ConstitutiveLaw::COMPUTE_STRESS );
 
+        // Set flags to only compute the stress
         r_flags.Set( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, true );
         r_flags.Set( ConstitutiveLaw::COMPUTE_STRESS, true );
 
-        // We compute the stress
-        ElasticIsotropic3D::CalculateMaterialResponseCauchy(rParameterValues);
+        ElasticIsotropic3D::CalculateMaterialResponsePK2(rParameterValues);
```

Comment:
why do you changes this?

Reply:
Idem

Reply:
I think that now, with the inclusion fo the initial strain, since the addition of this initial strain is done inside the calculateMaterialResponse, in order to print properly the strain is required to call the CalcMatResp....

Reply:
PK2 is the measure actually coded, so calling:  Cauchy -> PK2, which is OK inside ElasticIsotropic3D, but for derived CLs (e.g. Damage) it calls Elastic::CalculateMaterialResponseCauchy -> Damage::CalculateMaterialREsponsePK2. 

So trying to get the strain in the DamageCL makes it a recursive call:
Damage::CalculateMaterialPK2 -> Elastic::CalculateStrain -> Elastic::CalculateMaterialCauchy -> Damage::CalculateMaterialPK2...

This change (PK2 for Cauchy) looked the less invasive.